tests/Unit/API/ClientTest.php

```diff
@@ -56,7 +76,28 @@ protected function invoke_handler( string $handler_function ) {
 		$handler = new ReflectionMethod( GoogleServiceProvider::class, $handler_function );
 		$handler->setAccessible( true );
 
+		// Get access to internal container to share with service provider.
+		$woogle_container = new ReflectionClass( Container::class );
+		$league_container = $woogle_container->getProperty( 'container' );
+		$league_container->setAccessible( true );
+
 		$provider = new GoogleServiceProvider();
+		$provider->setLeagueContainer( $league_container->getValue( $this->container ) );
 		return $handler->invoke( $provider );
 	}
+
+	/**
+	 * Returns a mock client with an individual handler attached to the stack.
+	 *
+	 * @param string $handler_function Handler function name to include in stack.
+	 * @param array  $mocked_responses List of responses to return.
+	 *
+	 * @return Client Mock client.
+	 */
+	protected function mock_client_with_handler( string $handler_function, array $mocked_responses ) {
+		$mock     = new MockHandler( $mocked_responses );
+		$handlers = HandlerStack::create( $mock );
+		$handlers->push( $this->invoke_handler( 'error_handler' ) );
```

Comment:
I think it should be using the variable `$handler_function`.

Reply:
Nice catch, yes that's absolutely correct. Side effect of refactoring.